source/material_model/steinberger.cc

```diff
@@ -551,7 +580,15 @@ namespace aspect
                                   + " fields of type chemical composition."));
 
           has_background_field = (equation_of_state.number_of_lookups() == n_chemical_fields + 1);
-
+          std::vector<std::string> list_of_composition_names = this->introspection().get_composition_names();
+          if (equation_of_state.number_of_lookups() == 1)
+            list_of_composition_names.resize(1, "background");
+          if ((equation_of_state.number_of_lookups() == 1) && (has_background_field))
+            list_of_composition_names.resize(0, "background");
```

Comment:
Are you sure that these lines cover all the potential use cases of the material model? 

The way you have implemented this, if the number of lookups is equal to 1, the vector of composition names is replaced by  `{"background"}`. If additionally a background field is implied, the vector is empty `{}`.

What happens when the number of lookups is not equal to 1? You should test that `parse_map_to_double_array` does the correct thing in these cases.

You might be also be interested in mirroring the changes I made here, using the new map parser:
https://github.com/geodynamics/aspect/pull/5039/files#diff-cfe301f322bebee87432e53137677e6372743e5e142733d8fb63b5599c9322cb